### PR TITLE
Company data update endpoint

### PR DESCRIPTION
**Pull request by ebabic1**
Implemented PUT endpoint for updating tenants

### CODE REVIEW

File: backend/src/main/java/ba/unsa/etf/si/bbqms/tenant_service/implementation/TenantServiceImpl.java [hunk 1]
```diff
@@ -41,4 +42,26 @@ public Tenant addTenant(final AddTenantDto request){
     public Tenant findByCode(final String code) throws EntityNotFoundException {
         return tenantRepository.findByCode(code).orElseThrow(() -> new EntityNotFoundException("No tenant found with code: " + code));
     }
+
+    @Override
+    public Tenant updateTenant(final String code, final TenantDto request) throws EntityNotFoundException{
+        Tenant tenant = findByCode(code);
```
Comment (EmirKapic) [inline backend/src/main/java/ba/unsa/etf/si/bbqms/tenant_service/implementation/TenantServiceImpl.java]: ~~ `final`

File: backend/src/main/java/ba/unsa/etf/si/bbqms/ws/controllers/TenantController.java [hunk 2]
```diff
@@ -29,4 +29,13 @@ public ResponseEntity getTenantByCode(@PathVariable(name = "code") final String
             return ResponseEntity.badRequest().body(new ErrorResponseDto(e.getMessage()));
         }
     }
+
+    @PutMapping("/{code}")
+    public ResponseEntity updateTenant(@PathVariable String code, @RequestBody TenantDto request) {
```
Comment (EmirKapic) [inline backend/src/main/java/ba/unsa/etf/si/bbqms/ws/controllers/TenantController.java]: ~~ `final`

**Review by EmirKapic — DISMISSED** [review]

File: backend/src/main/java/ba/unsa/etf/si/bbqms/tenant_service/implementation/TenantServiceImpl.java (hunk 1 above)
Comment (ebabic1) [reply backend/src/main/java/ba/unsa/etf/si/bbqms/tenant_service/implementation/TenantServiceImpl.java]: Made tenant final

File: backend/src/main/java/ba/unsa/etf/si/bbqms/ws/controllers/TenantController.java (hunk 2 above)
Comment (ebabic1) [reply backend/src/main/java/ba/unsa/etf/si/bbqms/ws/controllers/TenantController.java]: Made params final

**Review by EmirKapic — APPROVED** [review]

**Pull request merged**